rfcs/0165-bootspecv2.md

```diff
@@ -0,0 +1,372 @@
+
+## Further iterations of Bootspec
+
+It has been brought that the RFC process may not be adequate to discuss further
+iterations of Bootspec.
+
+While the author disagree with this vision and believe this would be a loss
+for increased participation in the elaboration of further iterations and would lead to a 
+decrease in discoverability of Bootspec matters.
+
+We propose to move further iterations of Bootspec away from the RFC process.
+
+For this, we propose to follow a lightweight process inside a GitHub repository
+containing the specification and other relevant metadata about Bootspec.
+
+In instances of controversy or difficulties to reach consensus among the Bootspec ecosystem
+developers and maintainers, we should reach out to the standard RFC process.
+
+# Goals
+[goals]: #goals
+
+- Improve non-x86 support in Bootspec, emphasizing the importance of
+  devicetrees.
+- Address and reduce the risks associated with initrd secrets, providing a
+  default "secure" implementation within the systemd ecosystem and offering
+  flexibility for other ecosystems.
+- Enhance initrd flexibility, allowing developers to optimize their systems by
+  supporting multiple initrds, including one for microcode and other specific
+  purposes.
+- Make further iterations of Bootspec easier to build.
+
+### Non-Goals
+[non-goals]: #non-goals
+
+- Store TPM2-related information (hashes)
+  We believe that Bootspec is still too immature for this and pcrlock
+  (https://github.com/systemd/systemd/pull/28891) offers a more reliable and
+  robust solution for generating signed PCR policies.
+- Supporting SecureBoot.
+  Secure Boot has one maintained implementation that is being upstreamed: https://github.com/nix-community/lanzaboote
+  which was enabled by Bootspec v1.
+- Specifying how to discover generations. This is desirable, but should not be tied to bootspec directly since bootspec may be useful with diverse discovery mechanisms.
+- Address how bootloader backends should handle foreign bootloader data in their stead and manipulate it or how transition from a bootloader to another should happen.
+
+# Proposed Solution
+[proposed-solution]: #proposed-solution
+
+- `initrd` will be removed from the v2
+- `initrds` will be introduced as a list of initrd (compressed or uncompressed
+  CPIO archives), this list can be empty, but the field is **required**
+  nonetheless.
+- `org.nixos.initrd-secrets.v1` will now be an **official extension** of
+  Bootspec which should be an hashmap of strings (key) and strings (value)
+  where the key is the "name" of a secret and the value is the **filesystem path** towards a
+  secret.
+- `fdtdir` will be introduced as an opaque string to a directory in a shape as
+  the kernel outputs give them, e.g. U-Boot's extlinux directive `FDTDIR` is an
+  example of this behavior — it is **optional**
```

Comment:
```suggestion
- `fdtdir` will be introduced as an opaque string to a directory following the structure
   of the `dtbs` folder from the kernel, e.g. U-Boot's extlinux directive `FDTDIR` is an
  example of this behavior — it is **optional**
```

Alternatively to the `dtbs` folder, it could be phrased referencing the fact it's the `dtbs_install` target of the build.